src/Microsoft.Health.Fhir.Core/Features/Validation/ResourceProfileValidator.cs

```diff
@@ -60,18 +58,21 @@ public override IEnumerable<ValidationFailure> Validate(PropertyValidatorContext
                     var errors = _profileValidator.TryValidate(resourceElement.Instance);
                     foreach (var error in errors.Where(x => x.Severity == IssueSeverity.Error || x.Severity == IssueSeverity.Fatal))
                     {
-                        yield return new FhirValidationFailure(
+                        context.AddFailure(new FhirValidationFailure(
```

Comment:
>context.AddFailure [](http://example.com/codeflow?start=24&length=18)

Unfortunatelly it doesn't work that way :(


[https://docs.fluentvalidation.net/en/latest/custom-validators.html](https://docs.fluentvalidation.net/en/latest/custom-validators.html)

If you add Failure here it would try to add error message via 
GetDefaultMessageTemplate

I think we need to plumb this validators via Custom() method for RuleFor.

Reply:
Yeah you are right I noticed that as well. It has an extra error message per validator. I can look into how we can use Custom()